applications/FluidDynamicsApplication/custom_utilities/fluid_characteristic_numbers_utilities.cpp

```diff
@@ -65,11 +66,21 @@ namespace Kratos
         }
         element_vel /= static_cast<double>(n_nodes);
 
-        // Calculate element CFL
+        // Calculate element CFL for incompressible flows
         const double h_min = rElementSizeCalculator(r_geometry);
-        const double elem_cfl = norm_2(element_vel) * Dt / h_min;
 
-        return elem_cfl;
+        if(ConsiderCompressibility == false) {
+            return norm_2(element_vel) * Dt / h_min;
+        }
+
+        // Computing CFl for compressible flows
+        double sound_velocity = 0;
+        for (unsigned int i = 1; i < n_nodes; ++i) {
+            sound_velocity += r_geometry[i].GetValue(SOUND_VELOCITY);
+        }
```

Comment:
```suggestion
        double sound_velocity = r_geometry[0].GetValue(SOUND_VELOCITY);
        for (unsigned int i = 1; i < n_nodes; ++i) {
            sound_velocity += r_geometry[i].GetValue(SOUND_VELOCITY);
        }
```
Either this or starting the loop counter from 1.